### PR TITLE
Make offsets dynamic by default

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113694
* __->__ #113693

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by albanD — APPROVED** [review]

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -2011,9 +2011,7 @@ def _create_symbolic_sizes_strides_storage_offset(
 
         # TODO: make this configurable from outside policy; we made a policy
         # decision here where if all sizes are static, we are going to
-        # specialize all of the inner strides/offset too. We don't have to
-        # do this, and arguably we should ALWAYS allow for dynamic offset,
-        # this is cheap.
+        # specialize all of the inner stridest too.
         # TODO: This should be DYNAMIC, using DUCK for BC
         dynamic_strides_offset = DimDynamic.STATIC if all(r == DimDynamic.STATIC for r in dynamic_dims) else DimDynamic.DUCK
```
Comment (voznesenskym) [inline torch/fx/experimental/symbolic_shapes.py:2016]: rename to dynamic_strides

**Review by voznesenskym — APPROVED** [review]

**Comment by davidberard98** [comment]
I'm going to work through the test failures

**Comment by ezyang** [comment]
It's not as bad as I thought, but that activation checkpointing failure looks like cancer

**Comment by ezyang** [comment]
obsoleted by https://github.com/pytorch/pytorch/pull/113734

**Pull request closed without merging**